src/odemis/driver/tfsbc.py

```diff
@@ -0,0 +1,277 @@
+
+    def _openSerialPort(self, port):
+        if self._port == "/dev/fake":
+            return BeamControllerSimulator()
+        else:
+            return ModbusSerialClient(method='rtu', port=port,
+                                      baudrate=BAUDRATE, timeout=TIMEOUT,
+                                      stopbits=STOPBITS, parity=PARITY,
+                                      bytesize=BYTESIZE)
+
+    def _setShift(self, value):
+        """
+        :param value (float, float): x, y shift from the center (in m)
+        """
+        gain = self._metadata.get(model.MD_CALIB_SCALE)
+        if not gain:
+            raise ValueError("No ampere-to-meter conversion factor specified. Please run the alignment procedure "
+                             "first.")
+        offset = self._metadata.get(model.MD_CALIB_TRANSLATION)
+        if not offset:
+            logging.warning("No deflection offset specified, assuming 0 amps.")
+            offset = (0, 0)
+
+        rotation = self._metadata.get(model.MD_CALIB_ROTATION)
+        if not offset:
+            logging.warning("No rotation specified, assuming 0 rad.")
+            rotation = 0
+
+        # Convert to beam coordinate system (rotation)
+        value = (value[0] * math.cos(rotation) + value[1] * math.sin(rotation),
+                 -value[0] * math.sin(rotation) + value[1] * math.cos(rotation))
+
+        # Convert to numpy.array to simplify calculation
+        value = numpy.array(value, dtype=numpy.float64)
+        gain = numpy.array(gain, dtype=numpy.float64)  # amps / m
+        offset = numpy.array(offset, dtype=numpy.float64)  # amps
+        amp_range = numpy.array(RANGE_AMPS, dtype=numpy.float64)
+
+        # Convert to amperes (with gain)
+        value *= gain
+
+        # Add offset
+        value -= offset  # subtract offset (center amplitude)
+
+        # Convert to 16-bit integer
+        conversion_factor = 0xFFFF / (amp_range[1] - amp_range[0])
+        valuePos = (value - amp_range[0]) * conversion_factor
+        valueNeg = (-value - amp_range[0]) * conversion_factor
+        # Note: in the example script a constant of 0.5 is added before converting
+        # to int to always round up (why?)
+        register_values = [int(valueNeg[0]), int(valueNeg[1]), int(valuePos[0]), int(valuePos[1])]
+
+        # Read previous value of registers
+        ret = self._read_registers()
+        logging.debug("Register values before writing: %s." % ret)
+
+        self._write_registers(register_values)
+
+        # Read back values to see if everything went fine
+        ret = self._read_registers()
+        if ret != register_values:
+            logging.error("Register values after writing %s != %s." % (ret, register_values))
+
+        # Convert rounded shift back to meter for return value
+        value = (numpy.array(ret[2:]) / conversion_factor + amp_range[0] + offset) / gain * math.cos(rotation)
```

Comment:
Are you sure you can do that with rotation?! I think you need to use matrix multiplication (if you want to write on a single line).